packages/zevm-app-contracts/contracts/liquidity-incentives/RewardDistributorFactory.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.7;
+
+import "@openzeppelin/contracts/interfaces/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "@zetachain/zevm-protocol-contracts/contracts/system/SystemContract.sol";
+
+import "./RewardDistributor.sol";
+
+contract RewardDistributorFactory {
+    address public immutable zetaTokenAddress;
+    SystemContract private immutable systemContract;
+
+    mapping(uint256 => address) public incentivesContracts;
+    uint256 public incentivesContractsLen;
+
+    event RewardDistributorCreated(
+        address rewardDistributorContract,
+        address stakingToken,
+        address LPStakingToken,
+        address rewardToken,
+        address owner
+    );
+
+    constructor(address _zetaTokenAddress, address _systemContract) {
+        zetaTokenAddress = _zetaTokenAddress;
+        systemContract = SystemContract(_systemContract);
+    }
+
+    function createIncentive(
+        address owner,
+        ///@dev _rewardsDistribution is one who can set the amount of token to reward
+        address rewardsDistribution,
+        address stakingToken
+    ) public {
+        address LPTokenAddress = systemContract.uniswapv2PairFor(
+            systemContract.uniswapv2FactoryAddress(),
+            stakingToken,
+            zetaTokenAddress
```

Comment:
could this be abstracted to any token and we can input the zetaToken address when we call createIncentive? just thinking for the future if there may be some pools like zrc-20 stables that could receive incentives

Reply:
done!